src/Nirum/Parser.hs

```diff
@@ -162,13 +165,20 @@ uniqueName forwardNames label' = try $ do
     nameP :: Parser Name
     nameP = name <?> label'
 
-annotationArgumentValue :: Parser T.Text
+annotationArgumentValue :: Parser AnnotationArgument
 annotationArgumentValue = do
-    char '"'
-    value <- manyTill charLiteral (char '"')
-    return $ T.pack value
-
-annotationArgument :: Parser (Identifier, T.Text)
+    startQuote <- optional $ try $ char '"'
+    case startQuote of
+        Just _ -> do
+            v <- manyTill charLiteral (char '"')
+            return $ AText $ T.pack v
+        Nothing -> do
+            v <- many digitChar
+            case readMaybe v of
+                Just i -> return $ AInt i
+                Nothing -> fail "digit expected"
```

Comment:
We could define a more general parser to take one or more digits and returns `Integer`, i.e., `integer :: Parser Integer`, and then make this `annotationArgumentValue` to utilize that.